### DISCUSSION TITLE
ROW_NUMBER fails in Golang database/sql but works on DBeaver

### DISCUSSION THREAD

**Discussion by butaca** [planetscale/discussion#524] [Q&A]
Queries with ROW_NUMBER in Go using the database/sql package. The same queries succeed using DBeaver.

I'm thinking this is due to some setting in the connection string or something like that but I'm truly lost. I also found that some people are having the same issue with PlanetScale:
* https://github.com/prisma/prisma/issues/17122
* https://github.com/prisma/prisma/discussions/11746

The error I'm getting is: "Error 1105 (HY000): unknown error: syntax error...." but if I copy and paste the exact same query in DBeaver (against the same PlanetScale database) it works.

Any thoughts? Thanks! :)

**Comment by orware** [comment]
@butaca,

More than likely there is still some compatibility limitations on our end with utilizing `ROW_NUMBER` on the Vitess side of things.

Over in https://github.com/planetscale/discussion/discussions/290 I link to an existing Vitess issue that mentions that a bit along with a link to another discussion where I was able to make use of a very basic `ROW_NUMBER` example, but I would expect things more complex than that to have difficulty.

It would be nice if you can share the example query you may be using and compare that to the others in the Prisma issues you linked to in order to try and run some extra tests on our end if needed to gather some additional data.

**Comment by butaca** [comment]
@orware  Thanks for your response.

I found out that it fails when using a prepared statement, but success when using a regular query

```go
package main

import (
	"database/sql"
	"fmt"
	"os"

	_ "github.com/go-sql-driver/mysql"
)

func main() {
	dbstr := os.Getenv("DB_CONNECTION_STRING")
	if len(dbstr) == 0 {
		fmt.Println("DB_CONNECTION_STRING env var missing")
		os.Exit(1)
	}

	db, err := sql.Open("mysql", dbstr)
	if err != nil {
		panic(err)
	}

	creteTable := "CREATE TABLE `foo` (	`bar` tinyint(1) DEFAULT NULL ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;"
	// not handling errors
	db.Exec(creteTable)

	// ✅ this query succeeds
	query := "SELECT ROW_NUMBER() OVER () AS rn FROM foo"
	rows, err := db.Query(query)
	if err != nil {
		panic(err)
	}
	defer rows.Close()
	
	// ❌ db.Prepare fails
	stmt, err := db.Prepare(query)
	if err != nil {
		panic(err)
	}
	rows, err = stmt.Query()
	if err != nil {
		panic(err)
	}
	defer rows.Close()
	
}
```
